### PR TITLE
260: Zero Hour Contract Card

### PR DESCRIPTION
**Pull request by ChazUK**

### CODE REVIEW

File: src/app/features/workers/contract-with-zero-hours/contract-with-zero-hours.component.ts [hunk 1]
```diff
@@ -1,14 +1,88 @@
...
+  saveHandler(): Promise<WorkerEditResponse> {
+    return new Promise((resolve, reject) => {
+      const { zeroHoursContract } = this.form.controls;
+      this.messageService.clearError();
+
+      if (this.form.valid) {
+        const worker = this.worker || ({} as Worker);
+        worker.zeroHoursContract = zeroHoursContract.value;
+
+        this.subscriptions.push(this.workerService.setWorker(worker).subscribe(resolve, reject));
```
Comment (ChazUK) [inline src/app/features/workers/contract-with-zero-hours/contract-with-zero-hours.component.ts]: @wozzer72 API call made here to update Worker with parameter `zeroHoursContract`

File: src/app/features/workers/contract-with-zero-hours/contract-with-zero-hours.component.ts [hunk 2]
```diff
@@ -1,14 +1,88 @@
...
+      } else {
+        if (zeroHoursContract.errors && zeroHoursContract.errors.required) {
+          this.messageService.show('error', 'Please fill the required fields.');
```
Comment (wozzer72) [inline src/app/features/workers/contract-with-zero-hours/contract-with-zero-hours.component.ts]: Previous defects raised by Ann have asked for this message to be more specific; but there is also a separate initiative to start providing a given set of error messages (more user friendly than those typically written by a developer). It's a rock and a hard place.

File: src/app/features/workers/contract-with-zero-hours/contract-with-zero-hours.component.ts [hunk 3]
```diff
@@ -1,14 +1,88 @@
...
+    this.messageService.clearAll();
+  }
+
+  async submitHandler() {
+    try {
+      await this.saveHandler();
+
+      if (
+        this.worker.zeroHoursContract === 'Yes' ||
+        [Contracts.Agency, Contracts.Pool_Bank, Contracts.Other].includes(this.worker.contract)
```
Comment (wozzer72) [inline src/app/features/workers/contract-with-zero-hours/contract-with-zero-hours.component.ts]: A nice easy API this one. Have you proven this works against the deployed backend?

File: src/app/features/workers/contract-with-zero-hours/contract-with-zero-hours.component.ts [hunk 4]
```diff
@@ -1,14 +1,88 @@
...
+
+    this.workerId = this.workerService.workerId;
+
+    this.subscriptions.push(
+      this.workerService.getWorker(this.workerId).subscribe(worker => {
+        this.worker = worker;
+
+        if (worker.zeroHoursContract) {
+          this.form.patchValue({
+            value: worker.zeroHoursContract,
```
Comment (wozzer72) [inline src/app/features/workers/contract-with-zero-hours/contract-with-zero-hours.component.ts]: Have you been able to demonstrates this works by fetching data from the current deployed backend - but also when simply navigating back and forth using client only representation?

File: src/app/features/workers/contract-with-zero-hours/contract-with-zero-hours.component.html [hunk 5]
```diff
@@ -1,3 +1,24 @@
...
-</p>
+<div class="main-page-container col-12">
+
+  <form novalidate
+    (ngSubmit)="submitHandler()"
+    [formGroup]="form">
+
+    <header>
+      <h1 class="h1">Do they have a zero hours contract?</h1>
+    </header>
```
Comment (denny13) [inline src/app/features/workers/contract-with-zero-hours/contract-with-zero-hours.component.html]: No need for 'header' tag... this is a left over from Adam I am aware.

File: src/app/features/workers/contract-with-zero-hours/contract-with-zero-hours.component.ts [hunk 6]
```diff
@@ -1,14 +1,86 @@
...
+import { Contracts } from 'src/app/core/constants/contracts.enum';
 
 @Component({
   selector: 'app-contract-with-zero-hours',
-  templateUrl: './contract-with-zero-hours.component.html'
+  templateUrl: './contract-with-zero-hours.component.html',
 })
-export class ContractWithZeroHoursComponent implements OnInit {
+export class ContractWithZeroHoursComponent implements OnInit, OnDestroy {
+  public answersAvailable = ['Yes', 'No', `Don't know`];
```
Comment (denny13) [inline src/app/features/workers/contract-with-zero-hours/contract-with-zero-hours.component.ts]: I have noticed back ticks being used in place of single quotes in a few places.. any reason?

**Comment by ChazUK** [comment]
[Zero hours contract #152](https://trello.com/c/ZkXZ2BTz/260-zero-hours-contract-152)

**Pull request merged**